web-core-sdk/one-key-flow/src/App.tsx

```diff
@@ -0,0 +1,360 @@
+  apiKey: "AIzaSyB0nd9YsPLu-tpdCrsXn8wgsWVAiYEpQ_E",
+  authDomain: "web3auth-oauth-logins.firebaseapp.com",
+  projectId: "web3auth-oauth-logins",
+  storageBucket: "web3auth-oauth-logins.appspot.com",
+  messagingSenderId: "461819774167",
+  appId: "1:461819774167:web:e74addfb6cc88f3b5b9c92",
+};
+
+function App() {
+  const [web3auth, setWeb3auth] = useState<Web3AuthCore | null>(null);
+  const [usesTorus, setUsesTorus] = useState(false);
+  const [torus, setTorus] = useState<Torus | any>(null);
+  const [nodeDetailManager, setNodeDetailManager] = useState<NodeDetailManager | any>(null);
+  const [provider, setProvider] = useState<SafeEventEmitterProvider | null>(
+    null
+  );
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        // Initialising Web3Auth
+        const web3auth = new Web3AuthCore({
+          clientId,
+          chainConfig: {
+            chainNamespace: CHAIN_NAMESPACES.EIP155,
+            ...chainConfig,
+          },
+        });
+
+        const openloginAdapter = new OpenloginAdapter({
+          loginSettings: {
+            mfaLevel: "default",
+          },
+          adapterSettings: {
+            network: "testnet",
+            uxMode: "redirect",
+            loginConfig: {
+              jwt: {
+                name: "Web3Auth One Key Login Flow",
+                verifier,
+                typeOfLogin: "jwt",
+                clientId,
+              },
+            },
+          },
+        });
+        web3auth.configureAdapter(openloginAdapter);
+        setWeb3auth(web3auth);
+        await web3auth.init();
+
+        // Instantiating Torus for One Key Flow
+        const torus = new Torus({
+          enableOneKey: true,
+          network,
+        });
+        setTorus(torus);
+
+        const nodeDetailManager = new NodeDetailManager({ network, proxyAddress: CONTRACT_MAP[TORUS_NETWORK.TESTNET] });
+        setNodeDetailManager(nodeDetailManager);
+
+        if (web3auth.provider) {
+          setProvider(web3auth.provider);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    init();
+  }, []);
+
+  const signInWithGoogle = async (): Promise<UserCredential> => {
+    try {
+      const app = initializeApp(firebaseConfig);
+      const auth = getAuth(app);
+      const googleProvider = new GoogleAuthProvider();
+      const res = await signInWithPopup(auth, googleProvider);
+      console.log(res);
+      return res;
+    } catch (err) {
+      console.error(err);
+      throw err;
+    }
+  };
+
+  const login = async () => {
+    if (!web3auth) {
+      uiConsole("web3auth not initialized yet");
+      return;
+    }
+    // login with firebase
+    const loginRes = await signInWithGoogle();
+    // get the id token from firebase
+    const idToken = await loginRes.user.getIdToken(true);
+
+    // parse the id token to get the sub value - this will be verifier id (used to know the user)
+    const base64Url = idToken.split('.')[1]
```

Comment:
You can get useInfo from firebase with "sub" field rather thn splitting it like this, that would be more clean, Also left a comment about the significance of sub field.

Reply:
used the uid field from user - the value is same as sub